tests/meltano/core/test_project_settings_service.py

```diff
@@ -165,3 +170,16 @@ def test_warn_if_default_setting_is_used(self, subject, monkeypatch):
         ):
             with pytest.warns(RuntimeWarning):
                 subject.get(setting)
+
+    def test_meltano_settings_with_active_environment(
+        self, subject, monkeypatch, environment
+    ):
+        # make sure that meltano setting values are written to the root of `meltano.yml`
+        # even if there is an active environment
+        with monkeypatch.context() as mpc:
+            mpc.setattr(subject.project, "active_environment", environment)
+            assert subject.project.active_environment == environment
+            subject.set("database_max_retries", 10000)
+            value, source = subject.get_with_source("database_max_retries")
+            assert source == SettingValueStore.MELTANO_YML
+            assert value == 10000
```

Comment:
The `monkeypatch` fixture will automatically roll-back changes made after the test function returns. An explicit context is generally not required. See [the note about `monkeypatch.context` here](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html#global-patch-example-preventing-requests-from-remote-operations).

```suggestion
        monkeypatch.setattr(subject.project, "active_environment", environment)
        assert subject.project.active_environment == environment
        subject.set("database_max_retries", 10000)
        value, source = subject.get_with_source("database_max_retries")
        assert source == SettingValueStore.MELTANO_YML
        assert value == 10000
```